detect_current_range.cpp

```diff
@@ -0,0 +1,37 @@
+#include "detect_current_range.h"
+#include <iostream>
+#include <string>
+#include <vector>
+#include <cmath>
+
+using namespace std;
+
+vector<string> consecutiveCurrentRanges(uint32_t* currentInAmpere, uint32_t sizeOfcurrentInAmpere)
+{
+    uint32_t length = 1;
+    vector<string> rangeList;
+
+    if (sizeOfcurrentInAmpere != 0) {
+      std::sort(currentInAmpere, currentInAmpere + sizeOfcurrentInAmpere);
+
+      for(uint32_t counter = 1; counter <= sizeOfcurrentInAmpere; counter++) {
+        if ((counter == sizeOfcurrentInAmpere) || ((currentInAmpere[counter] - currentInAmpere[counter - 1]) > 1)) {

```

Comment:
currentInAmpere[counter] when counter = sizeOfcurrentInAmpere, tries to access beyond range element. Please check.
If it accesses beyond range element, then the return value could be garbage or undefined value or sometimes could lead to memory fault exception. So the design should carefully consider this.